develop/TOOLS/TOONuGet/Producing_NuGet.md

```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
+- NuGet needs to include a license file
+- NuGet needs to be signed as only signed packages can be published on nuget.org
+  - When creating the NuGet via [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will take care of signing and publishing.
 
-For example:
+### Project type
 
-![Assembly info in project settings](~/develop/images/Assembly_info_NuGet.png)
+Make use of the SDK project style. This gives the advantage of using dotnet tools in the pipeline and has the latest features. It also cleans up the csproj file which reduces potential merge conflicts.
 
-> [!TIP]
-> In case your source files have been documented with [XML documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/), make sure to set the [DocumentationFile](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/output#documentationfile) compiler option. You can set this option for a project in Visual Studio by selecting the project in Solution Explorer, right-clicking and selecting *Properties*. Then, from the *Build* tab, select the *XML Documentation File* checkbox. This will generate an XML file in addition to the assembly, which will be included in the NuGet package. Developers making use of your NuGet package will then be able to see this documentation in IntelliSense and the object browser.
+Using SDK-style projects, you can still target .NET Framework:
+```xml
+<PropertyGroup>
+    ...
+    <TargetFramework>netstandard2.0</TargetFramework> => <TargetFramework>net462</TargetFramework>
+    ...
+</PropertyGroup>
+```
 
-## Automatic NuSpec file creation
+### Dependencies
 
-The CI/CD pipeline will look for the presence of NuSpec files for the .NET Framework class library projects in the solution.
+If there is a dependency on a DataMiner DLL (from the Skyline DataMiner/Files folder), use the corresponding NuGet ([Skyline.DataMiner.Files.XXX](https://www.nuget.org/packages?q=Skyline.DataMiner.Files.)).  
+If it is not there, contact the [Data-Acquisition](mailto:support.data-acquisition@skyline.be) domain. They can then include this DLL in the automatic NuGet creation from DataMiner versions.  
 
-If no NuSpec file was found, the pipeline will automatically create one by extracting the required metadata from the AssemblyInfo.cs file of the project.
+If there is a dependency on multiple DataMiner DLLs that are by default included in a QAction/Exe, then make use of the DevPacks ([Skyline.DataMiner.Dev.XXX](https://www.nuget.org/packages?q=Skyline.DataMiner.Dev.))
 
-The pipeline expects the following to present in the AssemblyInfo.cs file:
+### Target Framework
 
-- The name of the assembly must start with `Skyline.DataMiner.`.
-- A description must be provided (with a maximum length of 4000 characters).
-- A copyright must be provided.
+If possible, try to target .NET Standard 2.0 as that gives the most flexibility to be used somewhere else.
```

Comment:
This is incorrect. We recently learned that NET Standard 2.0 doesn't actually work well unless you're using the very latest feature releases. I think we should continue making people target .NET Framework 4.6.2 for now.

Reply:
As discussed internally, targeting .NET 2.0 is fine. To make use of a .NET Standard NuGet in a QAction or Exe, you have to add a reference to netstandard. This is documented a bit below with the Target Framework section. I've changed it from a Note to Important to highlight it more.